### ISSUE TITLE
Final time period under observation in the logrank test.

### ISSUE THREAD

**Issue by denbonte** [CamDavidsonPilon/lifelines#1300]
Hey Cameron!

First of all I, as many others, greatly appreciate the time you spent (and are spending) working at the library - so thank you!

### TL;DR
`logrank_test`'s `t_0` is, in my opinion, doing something a bit counterintuitive - and could benefit at least from additional documentation.

### Issue
Coming to the subject of the issue. I spent some time trying to replicate in R the results a colleague obtained using lifelines. Being a user of both - and never having had any trouble with such replication step before (I usually do it as a sanity check) - I decided to look into this properly (so hopefully you will find this issue well documented).

The problem I had was with the `logrank_test` function. Now, in R `survfit` is used to compute the logrank p-val for different groups - and there is no such thing as `t_0`. So, as far as I know, the correct way to "cap" the survival analysis to a certain time limit is to simply set to zero the event flag for all the patients who survived more than the cap itself (basically, "faking" you simply don't know what happened next and thus considering not-censored patients for which an event occurred after the time cap).

Here's a dummy example to make things clearer:
```
# cap observation at, e.g., 6 years
data$event[which(data$time > 6)] = 0

fit <- survfit(Surv(time, event) ~ group, data = data)
pval_logrank = round(surv_pvalue(fit)[2], 4)
```

On the other hand, I noticed that some colleagues , when doing the exact same analyses using Lifelines instead of R, were simply setting the `t_0` argument of `statistics.logrank_test` to the aforementioned age cap. The reason they went for that instead of zeroing the event flag is that, well, in the documentation is stated that `t_0` is "the final time period under observation" - so I guess it could be someone assumes the event flag zeroing is what's actually happening behind the curtain.

Therefore, in short, I think it's reasonable to assume some people would expect to obtain the same logrank p-value obtained with the R snippet above by running:

```
logrank_test(group1["time"],
             group2["time"],
             group1["event"],
             group2["event"],
             t_0 = 6)
```

As I got different results from the two snippets, I decided to look into it, so I started from the `logrank_test` [function definition under statistics.py](https://github.com/CamDavidsonPilon/lifelines/blob/master/lifelines/statistics.py#L440). As already reported in the documentation, the function is implemented by means of the `multivariate_logrank_test` [function](https://github.com/CamDavidsonPilon/lifelines/blob/master/lifelines/statistics.py#L653) by setting the number of groups to two.

The `t_0` argument in the multivariate logrank function is then used as an argument for the `group_survival_table_from_events` [function found in the __init__.py file under utils](https://github.com/CamDavidsonPilon/lifelines/blob/cd9143a121e367aa2bf25b8908d5e8552af3ba09/lifelines/utils/__init__.py#L324).

Looking at the latter (line 423):

```
if int(limit) != -1:
    survival_table = survival_table.loc[:limit]
```

I realised that there is no censoring happening, but rather than the records of the patients for which the survival time exceeds `t_0` are simply dropped. Obviously, I managed to replicate my colleague results by:

- in lifelines, simply dropping those records and not setting `t_0`;
- in R, simply dropping those records.

Now, I don't think what is implemented in `group_survival_table_from_events` (and therefore when using the logrank functions in lifelines) is necessarily right or wrong - but I think it would be fair to say documenting the fact that all the records for which survival time exceeds `t_0` are in fact dropped (and therefore excluded completely from the analysis).

As you may imagine, in cases where the sample size is small (say N=200), this **can** make a huge difference (from p-val < 0.05 to a much higher p-val > 0.2). Equally obvious, in other cases (N much higher, or simply N~200 but with another survival time distribution) the difference is really hard to spot (so it was really down to luck we found out about this).

I'm pretty sure the way it's implemented makes complete sense to a fair share of people (potentially, all humanity but me and my colleague :-D) - but I hope this issue can spare easily avoidable errors to those who disagree!

**Comment by CamDavidsonPilon** [comment]
Hi @denbonte, this is a well-detailed issue - thank you! 

I'll spend some time digging into log rank tests again, and why I chose to drop records. For now, though, I'm going to categorize this as a r-lang difference, but I think it could actually be a bug.
